model/Bom.js

```diff
@@ -74,20 +75,26 @@ class Bom extends CycloneDXObject {
     // read-installed with default options marks devDependencies as extraneous
     // if a package is marked as extraneous, do not include it as a component
     if (pkg.extraneous) return
+    let component
     if (!isRootPkg) {
-      const component = new Component(pkg, this.includeLicenseText, lockfile)
+      component = new Component(pkg, this.includeLicenseText, lockfile)
       if (component.externalReferences === undefined || component.externalReferences.length === 0) {
         delete component.externalReferences
       }
       if (list[component.purl]) return // remove cycles
       list[component.purl] = component
     }
     if (pkg.dependencies) {
-      Object.keys(pkg.dependencies)
+      const deps = Object.keys(pkg.dependencies)
         .map(x => pkg.dependencies[x])
         .filter(x => typeof (x) !== 'string') // remove cycles
         .map(x => this.createComponent(x, list, lockfile))
+        .filter(x => !!x)
+      if (component && deps.length > 0) {
+        this.addDependency(new Dependency(component.bomRef, deps.map(({ bomRef }) => ({ ref: bomRef }))))
+      }
     }
+    return component
```

Comment:
new return type?
then please add the proper JSdoc return annotation.
see https://jsdoc.app/tags-returns.html

Reply:
Return type added.